caas/kubernetes/clientconfig/k8s_test.go

```diff
@@ -172,6 +173,67 @@ func (s *k8sConfigSuite) assertSingleConfig(c *gc.C) {
 		})
 }
 
+func StdinMockContextManager(content []byte, callBack func()) {
+	tmpfile, err := ioutil.TempFile("", "tmpfile")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	if _, err := tmpfile.Write(content); err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err := tmpfile.Seek(0, 0); err != nil {
+		log.Fatal(err)
+	}
+
+	oldStdin := os.Stdin
+	defer func() { os.Stdin = oldStdin }() // Restore original Stdin
+
+	os.Stdin = tmpfile
+	callBack()
+
+	if err := tmpfile.Close(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (s *k8sConfigSuite) TestGetKubeConfigFromStdin(c *gc.C) {
+	StdinMockContextManager(
+		[]byte(singleConfigYAML),
+		func() {
```

Comment:
Does all of this need to be in the callback func? You could limit it down to the ```clientconfig.K8SClientConfig()``` call couldn't you?

Reply:
sorry, what do u mean?
Here what was trying to achieve is like context manager in python.